### PR TITLE
Add inference benchmark option in run.py

### PR DESCRIPTION
**Pull request by kchygoe**
- [x] Add --bench and --trial option in run.py (in output dir)
- [x] Change print to logger
- [x] Run bench with .so file
- [x] Run bench with .pb file (with install tensorflow)
- [x] ~Run bench with .elf file~

## Motivation and Context

I want performance benchmark for inference-side devices

## Screenshots (if appropriate):
```
python run.py -m ../models/minimal_graph_with_shape.pb -c ../models/meta.yaml -i ../caltech101_classification_has_validation_20191010145425/export/save.ckpt-30/128x128/inference_test_data/raw_image.png --bench --trial=10
INFO:__main__:Beanchmark is triggered with 10 times trial
INFO:__main__:Function _pre_process took 0.001985342000000001 seconds to complete
INFO:__main__:Function _pre_process took 0.001781323999999973 seconds to complete
INFO:__main__:Function _pre_process took 0.0015847509999999954 seconds to complete
INFO:__main__:Function _pre_process took 0.0020573539999999557 seconds to complete
INFO:__main__:Function _pre_process took 0.0021465889999999765 seconds to complete
INFO:__main__:Function _pre_process took 0.0014984670000000033 seconds to complete
INFO:__main__:Function _pre_process took 0.0015263009999999522 seconds to complete
INFO:__main__:Function _pre_process took 0.002129837999999995 seconds to complete
INFO:__main__:Function _pre_process took 0.0018432219999999777 seconds to complete
INFO:__main__:Function _pre_process took 0.0017701199999999861 seconds to complete
INFO:__main__:Avg(func _pre_process): 0.0018323307999999817 sec.
INFO:__main__:Function _run took 1.2936168460000002 seconds to complete
INFO:__main__:Function _run took 0.186034539 seconds to complete
INFO:__main__:Function _run took 0.165938197 seconds to complete
INFO:__main__:Function _run took 0.16927644499999994 seconds to complete
INFO:__main__:Function _run took 0.181955125 seconds to complete
INFO:__main__:Function _run took 0.1702543460000001 seconds to complete
INFO:__main__:Function _run took 0.1659946790000002 seconds to complete
INFO:__main__:Function _run took 0.16076352299999952 seconds to complete
INFO:__main__:Function _run took 0.1720339179999999 seconds to complete
INFO:__main__:Function _run took 0.1715396349999998 seconds to complete
INFO:__main__:Avg(func _run): 0.28374072529999994 sec.
INFO:root:Output: (before post process)
[[0.1043328  0.10301837 0.10024849 0.09968052 0.09801209 0.09797116
0.09353667 0.10142121 0.09947895 0.10229976]]
INFO:__main__:Function _post_process took 0.0046468449999998995 seconds to complete
INFO:__main__:Function _post_process took 0.0013953019999997096 seconds to complete
INFO:__main__:Function _post_process took 0.0014312090000001554 seconds to complete
INFO:__main__:Function _post_process took 0.0013390870000002941 seconds to complete
INFO:__main__:Function _post_process took 0.0015897990000000028 seconds to complete
INFO:__main__:Function _post_process took 0.0015181779999999812 seconds to complete
INFO:__main__:Function _post_process took 0.001253987000000123 seconds to complete
INFO:__main__:Function _post_process took 0.0010709200000000862 seconds to complete
INFO:__main__:Function _post_process took 0.0014579610000002852 seconds to complete
INFO:__main__:Function _post_process took 0.0013487299999996871 seconds to complete
INFO:__main__:Avg(func _post_process): 0.0017052018000000223 sec.
INFO:root:Output: (after post process)
[[0.1043328  0.10301837 0.10024849 0.09968052 0.09801209 0.09797116
0.09353667 0.10142121 0.09947895 0.10229976]]
INFO:root:save json: output/output.json
INFO:root:save image: output/images/raw_image.png

```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature / Optimization (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: output_template/python/run.py [hunk 1]
```diff
@@ -100,29 +93,50 @@ def _run(model, input_image, config):
...
-    output = nn.run(data)
+    output = nn.run(image_data)
 
-    return output, raw_image
+    return output
+
+
+def _timerfunc(func, extraArgs, trial):
+    runtime = 0.
+    for i in range(0, trial):
```
Comment (iizukak) [inline output_template/python/run.py]: `range(0, trial):` -> `range(trial):` ?

File: output_template/python/run.py [hunk 2]
```diff
@@ -148,6 +162,61 @@ def run_prediction(input_image, model, config_file, max_percent_incorrect_values
     _save_images(output_dir, filename_images)
 
 
+def run_prediction_with_bench(input_image, model, config_file, max_percent_incorrect_values=0.1, trial=1):
```
Comment (iizukak) [inline output_template/python/run.py]: `run_prediction_with_bench` and `run_prediction` have similar lines.
Is it possible to unity?

File: output_template/python/run.py [hunk 3]
```diff
@@ -148,6 +162,61 @@ def run_prediction(input_image, model, config_file, max_percent_incorrect_values
...
+
+    # run the model to inference
+    output, bench_inference = _timerfunc(_run, (model, image_data, config), trial)
+
+    logging.info('Output: (before post process)\n{}'.format(output))
+
+    # pre process for output
+    output, bench_post = _timerfunc(_post_process, (output, config.POST_PROCESSOR), trial)
+
+    logging.info('Output: (after post process)\n{}'.format(output))
```
Comment (iizukak) [inline output_template/python/run.py]: Umm, I want `total` inference time. 
`total` means pre process + network inference + post process.
But not sure everyone wants.

**Comment by kchygoe** [comment]
@iizukak thx, I'll fix soon

File: output_template/python/run.py (hunk 3 above)
Comment (kchygoe) [reply output_template/python/run.py]: ok, how about adding `total` in bench output as sum of those pre,post,inference?

File: output_template/python/run.py (hunk 2 above)
Comment (kchygoe) [reply output_template/python/run.py]: Ya, I think so, let me try

File: output_template/python/run.py (hunk 3 above)
Comment (iizukak) [reply output_template/python/run.py]: @kchygoe sound nice.

File: output_template/python/run.py (hunk 2 above)
Comment (kchygoe) [reply output_template/python/run.py]: I wonder that making benchmark as a default behavior of run_prediction would be fine. How do you think?

Comment (kchygoe) [reply output_template/python/run.py]: @iizukak

Comment (iizukak) [reply output_template/python/run.py]: @kchygoe Yeah. I think so too.

File: output_template/python/run.py (hunk 1 above)
Comment (kchygoe) [reply output_template/python/run.py]: fixed

File: output_template/python/run.py (hunk 3 above)
Comment (kchygoe) [reply output_template/python/run.py]: added

**Comment by kchygoe** [comment]
I'll remove the --bench args

File: output_template/python/lmnet/utils/output.py [hunk 4]
```diff
@@ -159,6 +160,9 @@ def __call__(self, outputs, raw_images, image_files):
             "results": [],
         }
 
+        if self.bench is not None and type(self.bench) is dict:
```
Comment (tkng) [inline output_template/python/lmnet/utils/output.py]: I don't think it is necessary to guarantee the type of `self.bench` here. (I'm a little uncomfortable to care only `self.bench` specially.)

If you want to add a check mechanism, type annotation looks better way.

Comment (kchygoe) [reply output_template/python/lmnet/utils/output.py]: Fixed and rebased. could you please check?

**Pull request merged**